### PR TITLE
improve documentation for DsymbolTable

### PR DESCRIPTION
**Pull request by WalterBright**
Leave it better than I found it.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/dsymbol.d [hunk 1]
```diff
@@ -2186,41 +2186,61 @@ extern (C++) final class DsymbolTable : RootObject
...
+     * Params:
+     *   s = replacement symbol with same identifier
+     */
+    void update(Dsymbol s)
     {
-        const ident = s.ident;
-        Dsymbol* ps = tab.getLvalue(ident);
-        *ps = s;
-        return s;
+        *tab.getLvalue(s.ident) = s;
```
Comment (MoonlightSentinel) [inline src/dmd/dsymbol.d:2210]: Revert this, it's not an improvement to the documentation. 

(Also forgot to update dsymbol.h...)

**Review by MoonlightSentinel — CHANGES_REQUESTED** [review]

File: src/dmd/dsymbol.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/dsymbol.d:2210]: This PR is tagged "refactoring".

Updated dsymbol.h.

Comment (MoonlightSentinel) [reply src/dmd/dsymbol.d:2210]: A refactoring should not change the behavior... Have you checked that this change doesn't affect LDC?

Comment (WalterBright) [reply src/dmd/dsymbol.d:2210]: It doesn't change the behavior - dmd never calls it. As for gdc/ldc, if they're using the return value, it'll give an error, not silently corrupt anything. The return value makes no sense.

Comment (WalterBright) [reply src/dmd/dsymbol.d:2210]: It occurs to me you and I have different interpretations of what "refactoring" means. To me, it means not changing the behavior of the program. To you (correct me if I'm wrong) it means not changing the API of any functions.

Comment (MoonlightSentinel) [reply src/dmd/dsymbol.d:2210]: Refactorings should retain the behavior  of a component, it's more nuanced than the whole program but doesn't prohibit changes to function signatures.

The problem here is a change to the public API which can break code depending on "DMD as a library" - without providing any real benefit.

But I guess that is acceptable given that the potential breakage isn't silent and can easily be fixed.

**Review by MoonlightSentinel — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**